### PR TITLE
fix: generated redirect page canonical lacks of site prefix

### LINKED ISSUE #8499: generated redirect page canonical lacks of site prefix
### Astro Info

```block
Astro version            v2.10.15
Package manager          npm
Platform                 darwin
Architecture             arm64
Adapter                  astro-sst/edge
Integrations             @y_nk/cached-fetch, @astrojs/tailwind, @astrojs/react, @astrojs/sitemap
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

image

### What's the expected result?

canonical url should respect `import.meta.env.SITE` value.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/withastro-astro-aebx5j

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by rishi-raj-jain**
fix #8499

## Changes

Fix generated redirect page canonical lacks of site prefix

### CODE REVIEW

**Review by ematipico — COMMENTED** [review]
can we add a test please?

**Comment by rishi-raj-jain** [comment]
@ematipico Can you guide me on writing tests?

File: packages/astro/src/core/build/generate.ts [hunk 1]
```diff
@@ -488,7 +491,12 @@ function getUrlForPath(
 	return url;
 }
 
-async function generatePath(pathname: string, gopts: GeneratePathOptions, pipeline: BuildPipeline) {
+async function generatePath(
+	pathname: string,
+	gopts: GeneratePathOptions,
+	pipeline: BuildPipeline,
```
Comment (ematipico) [inline packages/astro/src/core/build/generate.ts:497]: The `pipeline` has already the information you need, you don't need to pass `siteURL` all around.

Just call `pipeline.getConfig()`

**Review by ematipico — CHANGES_REQUESTED** [review]
For a test, you could create a new test case inside `static-build.test.js`, and use the fixture `./fixtures/static-build/`.

Then it's just a matter of creating a new page to recreate the issue, and assert that the canonical attribute contains the expected string.

**Comment by rishi-raj-jain** [comment]
> For a test, you could create a new test case inside static-build.test.js, and use the fixture ./fixtures/static-build/.

Then it's just a matter of creating a new page to recreate the issue, and assert that the canonical attribute contains the expected string.

Thanks! That helped, pushed!

**Review by ematipico — APPROVED** [review]

**Pull request merged**